kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/KafkaProxyInitializerTest.java

```diff
@@ -201,6 +206,29 @@ void shouldAddCommonHandlersOnBindingComplete(boolean tls) {
         verifyer.verify(channelPipeline).addLast(eq("netHandler"), any(KafkaProxyFrontendHandler.class));
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    void shouldRemoveAndReAddErrorHandlerOnBindingComplete(boolean tls) {
```

Comment:
I'm not sure if its a separate test or part of this one, but I think we are missing an assertion that the logging error handler is at the tail of pipeline. e.g. nothing called `addLast` after it was added

Reply:
Have pushed up a refactor to make all the tests care about the order the handlers are added in. I don't think the existing tests like "if I enable network logging, then a logging handler is added" are that useful since network logging won't work if the handler is in the wrong order.